pkg/ebpf/tracer.go

```diff
@@ -126,6 +128,19 @@ func NewTracer(config *Config) (*Tracer, error) {
 		}
 	}
 
+	var reverseDNS ReverseDNS = nullReverseDNS{}
+	if config.DNSSnooping {
+		filter := m.SocketFilter("socket/dns_filter")
+		if filter == nil {
+			return nil, fmt.Errorf("error retrieving socket filter")
+		}
+
+		reverseDNS, err = NewSocketFilterSnooper(filter)
+		if err != nil {
+			return nil, fmt.Errorf("error creating dns snooper: %s", err)
```

Comment:
I recommend logging and falling back to nullSocketFilter in case of an error. We did something similar for conntrack and it was helpful when rolling out a new system-probe release while ironing out issues with conntrack.

Reply:
+1 for this. Let's not fail hard if dns inspection fails

Reply:
will do 👍